### PR TITLE
gdcm,vtk@8.2: migrate to python@3.10

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

### CODE REVIEW

**Comment by carlocab** [comment]
Seems to crash when it tries to run `vtkpython`.

**Comment by cho-m** [comment]
I was looking into `vtk` and hit same issue, which I found was lack of Python 3.10 support in current stable release. At least for `vtk` v9, there is a fix https://gitlab.kitware.com/vtk/vtk/-/merge_requests/8738 that can either be patched or will be available in 9.2.0.

We could try to backport it to v8

**Comment by carlocab** [comment]
I don't mind keeping it on `python@3.9`. But if you're keen to backport the patch, I'm okay with that too.

**Comment by cho-m** [comment]
I also think `python@3.9` is fine given that we are trying to re-deprecate (and ultimately remove) `vtk@8.2`

I would like to get `vtk@8.2` migrated to `jpeg-turbo` since that should be a drop-in. Maybe at same time as `itk`.

---

For current `vtk`, I am looking into backporting Python 3.10 patch with something like following, but need to check how bad dependency tree migration is (e.g. `opencv`, `gdal`, ...). Particularly trying to avoid pulling in formulae with `gobject-introspection`, if possible.
```rb
    # Fix vtkpython support for Python 3.10. Remove in the next release.
    # First patch backports part of older commit so we can directly patch in upstream commit.
    patch :DATA
    patch do
      url "https://gitlab.kitware.com/vtk/vtk/-/commit/3eea0e12acfb608a76d6ae36fb36566a4a6b0e9b.diff"
      sha256 "1c1c4622a58f8c852d196759c8d9036e4d513a5ebe16fe0bfa14583832886572"
    end

...

__END__
diff --git a/Documentation/release/dev/python-3.10-wheels.md b/Documentation/release/dev/python-3.10-wheels.md
new file mode 100644
index 0000000000000000000000000000000000000000..f4e81411c73f30724ad420ccb7f3c6c07a6f8e3d
--- /dev/null
+++ b/Documentation/release/dev/python-3.10-wheels.md
@@ -0,0 +1,7 @@
+## Python 3.10 wheels
+
+VTK now generates Python 3.10 wheels. Note that `vtkpython` and other tools
+using `vtkPythonInterpreter` still do not support the new initialization
+behaviors introduced in Python 3.10. See [this issue][vtk-python-3.10-support].
+
+[vtk-python-3.10.support]: https://gitlab.kitware.com/vtk/vtk/-/issues/18317
```

File: Formula/gdcm.rb [hunk 1]
```diff
@@ -25,7 +26,7 @@ class Gdcm < Formula
   depends_on "swig" => :build
   depends_on "openjpeg"
   depends_on "openssl@1.1"
-  depends_on "python@3.9"
+  depends_on "python@3.10"
   depends_on "vtk@8.2"
```
Comment (cho-m) [inline Formula/gdcm.rb:30]: May be worth checking what features we lose by dropping VTK support. Fedora and AUR (and maybe FreeBSD, not sure of DSL they use) has disabled VTK. NixOS defines VTK as optional.

Comment (cho-m) [reply Formula/gdcm.rb:30]: Opened #107657 for this approach

**Comment by branchvincent** [comment]
Closing in favor of #107657

**Pull request closed without merging**